app/src/main/java/ml/docilealligator/infinityforreddit/markdown/SpoilerParserPlugin.java

```diff
@@ -159,28 +155,57 @@ private ArrayList<Pair<Integer, Integer>> parse(SpannableStringBuilder markdown,
                     && (currentChar != '<')
                     && (currentChar != '!')) {
                 new_lines = 0;
-            } else if ((i + 1 < length)
-                    && currentChar == '>'
-                    && markdown.charAt(i + 1) == '!') {
-                openSpoilerStack.push(i + 2);
-            } else if ((i + 1 < length) && (i - 1 >= 0)
-                    && openSpoilerStack.size() > 0
-                    && markdown.charAt(i - 1) != '>'
+            } else if (currentChar == '>'
+                    && i + 1 < length
+                    && markdown.charAt(i + 1) == '!'
+                    && noCodeIntersection(markdown, i)) {
+                openSpoilerStack.push(i);
+                i++; // skip '!'
+            } else if (openSpoilerStack.size() > 0
                     && currentChar == '!'
-                    && markdown.charAt(i + 1) == '<') {
+                    && i + 1 < length
+                    && markdown.charAt(i + 1) == '<'
+                    && noCodeIntersection(markdown, i)) {
                 var pos = openSpoilerStack.pop();
-                if (!closedSpoilers.isEmpty()
-                        && closedSpoilers.get(closedSpoilers.size() - 1).first > pos
-                        && closedSpoilers.get(closedSpoilers.size() - 1).second < i) {
-                    closedSpoilers.remove(closedSpoilers.size() - 1);
-                }
-                if (pos != i) {
-                    closedSpoilers.add(Pair.create(pos, i));
+                while (!spoilersStack.isEmpty()
+                        && spoilersStack.peek().start > pos) {
+                    SpoilerRange nestedRange = spoilersStack.pop();
+                    nestedRange.nested = true;
+                    closedSpoilers.add(nestedRange);
```

Comment:
Because nested spoilers were just forgotten we could not remove their tags. Fixes
```
>! one >! two !< three !<
```